clients/src/main/java/org/apache/kafka/common/record/RecordBatch.java

```diff
@@ -55,6 +55,12 @@
     int NO_SEQUENCE = -1;
 
     /**
+     * For offset commit messages, we need to skip the sequence checks since sequence numbers are not
+     * generated on the producer for these messages.
+     */
+    int SKIP_SEQUENCE_CHECK = -3;
```

Comment:
A bit unfortunate that we need this. If we want to support this, we'll have to add something to `LogValidator` to ensure that users aren't using this as well. One alternative that comes to mind is to add an option to `Log` to disable the sequence check if it's an internal topic (or maybe just if it's __consumer_offsets). Then we could use `NO_SEQUENCE`. Seems a bit nicer to keep this bit of ugliness in the code and not the protocol.